src/app/content/components/ContentLink.tsx

```diff
@@ -71,6 +77,15 @@ export const ContentLink = (props: React.PropsWithChildren<Props>) => {
       if (onClick) {
         onClick();
       }
+
+      const searchParams = queryString.parse(assertWindow().location.search);
+      if (searchScrollTarget) {
+        searchParams.target = JSON.stringify(omit('elementId', searchScrollTarget));
+      }
+      const options = searchScrollTarget
+        ? { hash: searchScrollTarget.elementId, search: queryString.stringify(searchParams) }
```

Comment:
I've modified it to always pass `search.query` and if `scrollTarget` is passed as prop (direct prop on the component, not from the state) then it will also set `target` and `hash` from this `scrollTarget`